HW15/src/FileLogger.java

```diff
@@ -0,0 +1,32 @@
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class FileLogger {
+
+    private final FileLoggerConfiguration fileLoggerConfiguration;
+
+
+    public FileLogger(){
+        fileLoggerConfiguration = new FileLoggerConfiguration();
+    }
+
+
+    public void info(String message) throws IOException, FileMaxSizeReachedException {
+        if(!fileLoggerConfiguration.isMaxFileSizeExceeded()) {
+            writeLog(message, FileLoggerConfiguration.LoggingLevel.INFO);
+        }
+    }
+
+    public void debug(String message) throws IOException, FileMaxSizeReachedException {
+        if(fileLoggerConfiguration.isMaxFileSizeExceeded()) {
+            writeLog(message, FileLoggerConfiguration.LoggingLevel.INFO);
+            writeLog(message, FileLoggerConfiguration.LoggingLevel.DEBUG);
+        }
+    }
+
+    private void writeLog(String message, FileLoggerConfiguration.LoggingLevel loggingLevel) throws IOException {
+        try (FileWriter logsFileWriter = new FileWriter(fileLoggerConfiguration.getFile(), true)) {
```

Comment:
Each time you write a log - you open a file (i.e. get a resource) and close this resource at the end of the method. Maybe it is better to open once and close after? Not sure about it, but think about it.